### PR TITLE
[enhancement] use react-toast-notifications for qr code scanning

### PR DESCRIPTION
**Pull request by WilhelmWillie**
Kill the jank thing I was doing to show feedback, instead use `react-toast-notifications` 

Thank you, abstractions!

### CODE REVIEW

File: api/live.js [hunk 1]
```diff
@@ -176,6 +176,10 @@ router.post("/dispatch", async (req, res) => {
 async function handleContrib(userId, req, res) {
   const input = req.body;
 
+  const profile = await models.HackerProfile.findOne({
```
Comment (jzeiders) [inline api/live.js]: Can use models.HackerProfile.findByPK(userId)

File: api/live.js [hunk 2]
```diff
@@ -217,6 +224,26 @@ async function handleEmailContrib(userEmail, req, res) {
 
 async function handleCheckin(userId, req, res) {
   try {
+    const profile = await models.HackerProfile.findOne({
```
Comment (jzeiders) [inline api/live.js]: Same PKEY

File: api/live.js [hunk 3]
```diff
@@ -217,6 +224,26 @@ async function handleEmailContrib(userEmail, req, res) {
...
+      where: { userId: userId }
+    });
+    const profileStatus = profile.get("status");
+    const invalidStatuses = [
```
Comment (jzeiders) [inline api/live.js]: Do we have an enum statuses somewhere? This many magic strings scare me

File: pages/scan.tsx [hunk 4]
```diff
@@ -1,111 +1,110 @@
...
-        setSuccessfulScan({
-          firstName: idData.firstName,
-          lastName: idData.lastName
-        });
+    // Send a request to the server to scan hacker for task
+    const scanRequest = await fetch("/api/live/dispatch", {
+      method: "POST",
+      body: JSON.stringify({
+        userId: code,
+        actionId: action === "checkin" ? "checkin" : "contrib",
```
Comment (jzeiders) [inline pages/scan.tsx]: Another enum chance, especially since it's TS. Can just to odyssey.d.ts

Comment (WilhelmWillie) [reply pages/scan.tsx]: yeah, i'm probably going to address some of this in a later PR -- mostly wrote this to do a quick test for doing QR scans for both checkin and contributions

File: api/live.js (hunk 3 above)
Comment (WilhelmWillie) [reply api/live.js]: we don't, prob one of the regrets i have is not having some constant file early on, tho i can address in a follow up

**Pull request merged**